opencog/openpsi/main.scm

```diff
@@ -134,15 +140,36 @@
 )
 
 ; -------------------------------------------------------------
-(define (psi-halt component)
+(define (save-thread thread component)
+ "
+  save-thread THREAD COMPONENT
+
+  Save thread in the psi-threads list to properly terminate it
+  before exit.
+ "
+ (set! psi-threads (cons (cons thread component) psi-threads))
+)
+
+; -------------------------------------------------------------
+(define* (psi-halt component #:optional (timeout 3))
```

Comment:
I'm thinking that an infinite default timeout would be better. That would be guaranteed to always have the correct behavior (although it might hang if there are other bugs).